pkg/kcp/provider/aws/mock/elastiCacheClientFake.go

```diff
@@ -45,38 +45,28 @@ func getDefaultParams() map[string]elasticacheTypes.Parameter {
 }
 
 type elastiCacheClientFake struct {
-	subnetGroupMutex    *sync.Mutex
-	parameterGroupMutex *sync.Mutex
-	elasticacheMutex    *sync.Mutex
-	secretStoreMutex    *sync.Mutex
-	userGroupsMutex     *sync.Mutex
-	securityGroupsMutex *sync.Mutex
-	replicationGroups   map[string]*elasticacheTypes.ReplicationGroup
-	cacheClusters       map[string]*elasticacheTypes.CacheCluster
-	parameters          map[string]map[string]elasticacheTypes.Parameter
-	parameterGroups     map[string]*elasticacheTypes.CacheParameterGroup
-	subnetGroups        map[string]*elasticacheTypes.CacheSubnetGroup
-	userGroups          map[string]*elasticacheTypes.UserGroup
-	secretStore         map[string]*secretsmanager.GetSecretValueOutput
-	securityGroups      []*ec2Types.SecurityGroup
+	mutex             *sync.Mutex
+	replicationGroups map[string]*elasticacheTypes.ReplicationGroup
+	cacheClusters     map[string]*elasticacheTypes.CacheCluster
+	parameters        map[string]map[string]elasticacheTypes.Parameter
+	parameterGroups   map[string]*elasticacheTypes.CacheParameterGroup
+	subnetGroups      map[string]*elasticacheTypes.CacheSubnetGroup
+	userGroups        map[string]*elasticacheTypes.UserGroup
+	secretStore       map[string]*secretsmanager.GetSecretValueOutput
+	securityGroups    []*ec2Types.SecurityGroup
 }
 
 func newElastiCacheClientFake() *elastiCacheClientFake {
 	return &elastiCacheClientFake{
-		elasticacheMutex:    &sync.Mutex{},
-		subnetGroupMutex:    &sync.Mutex{},
-		parameterGroupMutex: &sync.Mutex{},
-		secretStoreMutex:    &sync.Mutex{},
-		userGroupsMutex:     &sync.Mutex{},
-		securityGroupsMutex: &sync.Mutex{},
-		replicationGroups:   map[string]*elasticacheTypes.ReplicationGroup{},
-		cacheClusters:       map[string]*elasticacheTypes.CacheCluster{},
-		subnetGroups:        map[string]*elasticacheTypes.CacheSubnetGroup{},
-		parameterGroups:     map[string]*elasticacheTypes.CacheParameterGroup{},
-		parameters:          map[string]map[string]elasticacheTypes.Parameter{},
-		secretStore:         map[string]*secretsmanager.GetSecretValueOutput{},
-		userGroups:          map[string]*elasticacheTypes.UserGroup{},
-		securityGroups:      []*ec2Types.SecurityGroup{},
+		mutex:             &sync.Mutex{},
```

Comment:
why not `m sync.Mutex`?

Reply:
I dont like non-descriptive names

Reply:
I was actually pointing to both, name and the type. I was curious why pointer. I looked at the https://pkg.go.dev/sync#Mutex and `func (m *Mutex) Lock()` operates over pointer but I couldn't find reason why we should use `*sync.Mutex` and not `sync.Mutex` except that other implantations, [among go tour](https://go.dev/tour/concurrency/9), are using `sync.Mutex`. Anyway, not a big deal except that we should try to be consistent.

Reply:
I will change it to value